core/domain/state_domain.py

```diff
@@ -2336,6 +2336,36 @@ def create_default_subtitled_unicode(cls, content_id):
         return cls(content_id, '')
 
 
+class TranslatableItem(python_utils.OBJECT):
+    """Value object representing item that can be translated."""
+
+    DATA_FORMAT_HTML = 'html'
+    DATA_FORMAT_UNICODE_STRING = 'unicode'
+    DATA_FORMAT_SET_OF_NORMALIZED_STRING = 'set_of_normalized_string'
+    DATA_FORMAT_SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+    def __init__(self, content, data_format):
+        """Initializes a SubtitledUnicode domain object.
+
+        Args:
+            content: str. The translatable content text.
```

Comment:
The content could also be a list of strings if it corresponds to TextInput or SetInput.

Reply:
@sagangwee  do we actually allow TextInput and SetInput rules to be translated via the contributor dashboard as of now? I added rules for these interactions but could not find them in the contributor dashboard.

Also from what I can see, it seems like the function that fetches content to be translated does not currently handle translation of rules.

https://github.com/oppia/oppia/blob/a13aa30b1042006075dcb1e027d4981953578657/core/domain/state_domain.py#L3071-L3125

I might be missing something. Can you please confirm? Thanks!

Reply:
Ah, sorry for the confusion. Yes you are correct that the contributor dashboard does not currently support translation of rules. I just figured you might want to align the field conceptually with what we have with the WrittenTranslation class as this domain object is not specifically tied to the contributor dashboard. And we might as well future-proof the field now as rules translation is coming very soon, but either way would be fine for now.

Reply:
Oh I see. Thanks for explaining. I've fixed the doc string now.